import "User.proto";

package ProtoShakeDice;

//摇骰子活动
message ShakeDiceCPP
{
	required uint32 version	= 1;  //活动版本号
	required uint32 curmap	= 2;  //当前地图
	required uint32 map1_status	= 3;  //地图1状态
	required uint32 map2_status	= 4;  //地图2状态
	required uint32 map3_status	= 5; //地图3状态
	required uint32 dice_times	= 6; //已用摇骰子次数
	required uint32 reset_times	= 7; //已用重置次数
	required uint32 steps		= 8;  //可用步数
	required uint32 operate_ts	= 9; //摇骰子或重置的操作时间
	required uint32 extra_reward_status	= 10; //额外奖励状态
	required uint32 map1_rewards = 11;  //地图1奖励领取状态	
	required uint32 map2_rewards = 12;  //地图2奖励领取状态	
	required uint32 map3_rewards = 13;  //地图3奖励领取状态	
	required uint32 curpos		 = 14;  //英雄当前位置
}

//打开窗口协议
message OpenShakeDiceWindowReq
{
	
}

message OpenShakeDiceWindowResp
{
	required ShakeDiceCPP dice	= 1;
}

//摇骰子
message RollDiceReq
{

}

message RollDiceResp
{
	required ShakeDiceCPP dice	= 1;
	required uint32 cash		= 2;
}

//重置地图
message ResetMapReq
{

}

message ResetMapResp
{
	required ShakeDiceCPP dice	= 1;   //活动信息
	required uint32 cash		= 2;  //钻石数目
}

//进入地图
message EnterMapReq
{
	required uint32 mapid	= 1; //地图id
}

message EnterMapResp
{
	required ShakeDiceCPP dice	= 1;  //活动信息
}

//解锁地图
message UnlockMapReq
{
	required uint32 mapid = 1;
} 

message UnlockMapResp
{
	required ShakeDiceCPP dice	= 1;
}

//位置奖励
message PositionRewardReq
{
	repeated uint32 pids	= 1;  //经过的点
}

message PositionRewardResp
{
	required ShakeDiceCPP dice	= 1;
	required User.RewardBagInfo reward	= 2;  //奖励
}

//额外奖励
message ReceiveExtraRewardReq
{
	
}

message ReceiveExtraRewardResp
{
	required ShakeDiceCPP dice	= 1;
	required User.RewardBagInfo reward	= 2;  //奖励
}

